apps/avatax/src/modules/saleor/order-parser.ts

```diff
@@ -14,3 +15,28 @@ export class SaleorOrderParser {
     return safeParse(payload);
   }
 }
+
+export class SaleorOrderCancelledParser {
+  public static ParsingError = BaseError.subclass("AvataxAppSaleorOrderCancelledParsingError");
+
+  public static parse(payload: unknown) {
+    const verifiedPayload = verifyOrderCanceledPayload(payload);
+
+    if (verifiedPayload.isErr()) {
+      throw verifiedPayload.error;
+    }
+
+    const safeParse = Result.fromThrowable(
+      SaleorCancelledOrder.parser,
```

Comment:
I dont see why we have static method `parser` in  `SaleorCancelledOrder` and separate class called `SaleorOrderCancelledParser` with method `parse` that calls `SaleorCancelledOrder.parser`

Lets revisit this